app.rb

```diff
@@ -0,0 +1,83 @@
+require_relative 'person'
+require_relative 'book'
+require_relative 'rental'
+require_relative 'teacher'
+require_relative 'student'
+
+class App
+  attr_reader :people, :books, :rentals
+
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_all_books
+    puts 'All Books:'
+    @books.each do |book|
+      puts "Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  def list_all_people
+    puts 'All People:'
+    @people.each do |person|
+      puts "ID: #{person.id}, Name: #{person.name}, Age: #{person.age}"
+    end
+  end
+
+  def create_person(name, age, type, specialization = nil, classroom = nil)
+    if type == 'teacher'
+      person = Teacher.new(age, specialization, name: name)
+    elsif type == 'student'
+      person = Student.new(age, classroom, name: name)
```

Comment:
Kindly make sure that for the Person creation.

- Creating a student there's an option `Has parent permission`.
- Creating a teacher there's an option to add `specialization` as it is illustrated in the example video.

![image](https://github.com/karanJ2212/School_library_ruby/assets/38088818/66f260c4-36b8-446c-8e72-dbed54cb7ee7)